packages/core/src/types.ts

```diff
@@ -852,7 +852,7 @@ export interface MiddlewareConfig {
 export interface ApiClientFactoryParams<T, F = any> {
   api: F;
   isProxy?: boolean;
-  onCreate: (config: T, headers?: Record<string, string>) => { config: T; client: any };
+  onCreate: (config: T, headers?: Record<string, string>) => Promise<{ config: T; client: any }>;
```

Comment:
I know it's not a scope of this task. But maybe we should remove `headers` argument as we already changing smth here? I do not see any place where it's used. Do I miss something?

Reply:
No, you are not missing anything, it looks like an old code or a dead code. I will remove it then.